pkg/trace/traceutil/azure.go

```diff
@@ -144,6 +156,12 @@ func getLinuxRuntime(getenv func(string) string) (rt string) {
 	return rt
 }
 
+func inFunctionApp() bool {
+	_, existsRuntime := os.LookupEnv("FUNCTIONS_WORKER_RUNTIME")
+	_, existsVersion := os.LookupEnv("FUNCTIONS_EXTENSION_VERSION")
```

Comment:
I'm not sure I understand why `getAppServicesTags` accepts a getenv function instead of using `os.Getenv` directly in that function, but given that it does work this way, it's a bit odd that we're now using `os.LookupEnv` here. (We don't do that anywhere else in this file)

Do you need to do an explicit lookup because it's possible that `FUNCTIONS_WORKER_RUNTIME` and `FUNCTIONS_EXTENSION_VERSION` could have `""` as a valid value, and you need to decipher between a **default** empty string versus an **explicitly set** empty string? I don't see a test verifying that, so presumably it's not something that needs to be handled? If it is, please add a test.

If you don't need to do that, then I recommend just relying on `getenv` as this file does elsewhere. That would also eliminate the need for this helper function where we have to look up the same env more than once (which as I commented below, could have other issues)

Reply:
When I wrote this file originally it was suggested to pass `getAppServicesTags` the `getenv` function so that it was easier to mock the variables for tests. If that's not necessary we can definitely refactor it.

I only used the lookup because that's what we do [here](https://github.com/DataDog/datadog-agent/blob/main/pkg/trace/config/config.go#L596-L599). These environment variables should not be present unless in the case of a running Azure function in an Azure App Service environment, but I can refactor that as well.

Thanks for your advice!

Reply:
> When I wrote this file originally it was suggested to pass getAppServicesTags the getenv function so that it was easier to mock the variables for tests. If that's not necessary we can definitely refactor it, but I'd like to get this bug fix out first so we can get the UI issues sorted if that's OK.

It shouldn't be necessary, since you can either set the env with `os.SetEnv` in the tests, or with `t.SetEnv` in the tests, so that should make things more straightforward in the code.

> With that in mind, would it be better to move this to the config and have another bool like we do here?

That sounds reasonable! Whatever makes the most sense, I'll let you use your best judgement.